pkg/controller/controllercmd/builder.go

```diff
@@ -182,34 +185,32 @@ func (b *ControllerBuilder) Run(config *unstructured.Unstructured, ctx context.C
 		}
 	}
 
-	switch {
-	case b.servingInfo == nil && len(b.healthChecks) > 0:
+	if b.servingInfo == nil && len(b.healthChecks) > 0 {
```

Comment:
I think just require serving info overall.  I see no reason to have an option without it.